### ISSUE TITLE
Where to save binary data files

### ISSUE THREAD

**Issue by timothyas** [ECCO-GROUP/ECCOv4-py#72]
This package requires `binary_data/basins.data` for the `get_basin` module. Also, the (currently limited) test suite requires some binary data to test array conversion. As addressed by @ifenty in #70, keeping this directory as `bin/` was not a good idea because installing the package through `pip install` can result in these data files getting written to the installing user's `$HOME/bin` directory. 

I just tried installing the package through pip: it recognized/installed in my anaconda directory and installed as follows: 

anaconda3/lib/python3.6/site-packages/ecco_v4_py
anaconda3/ecco_v4_py_binary_data/basins.(meta/data) 

This would break `get_basin.py` too, since it expects the basins.data file to be in `../binary_data/basins.data` (along with the tests). I am now understanding #59 much better. I am curious @cspencerjones - where did you end up finding `basins.data`? 

TL; DR we need a consistent method for storing the few data files we need. Perhaps [figshare](https://figshare.com/) is the way to go, and I'll start by checking this out. Anyone have any experience with doing this in a standard way?

**Comment by ifenty** [comment]
Current behavior is as follows:
1: if the package is downloaded from github then the binary files are stored in 'ecco-v4-py/binary_data'.  
2: if the package is installed using pip, then the binary files are stored in '$HOME/binary_data'.  

Uploading the files onto figshare was relatively painless:
https://figshare.com/articles/Binary_files_for_the_ecco-v4-py_Python_package_/9932162

Now the code has error messages that direct a user to figshare for the binaries.

@timothyas I updated 'get_basin.py' but I don't know about the tests you are referring to.  Are there other places where 'the ./bin' directory was hard coded?

**Comment by cspencerjones** [comment]
That explains it! In the end I downloaded it manually from the directory `bin/` on github. Ian's solution seems good to me.

**Comment by timothyas** [comment]
Thanks @ifenty for implementing that so quickly. The module `ecco_v4_py/test/test_llc_array_conversion.py` relies on those files being there. This is more for internal testing so I'm less worried about it, but I can add a print statement there too.

**Comment by ifenty** [comment]
@timothyas I don’t actually see what you’re talking about in test_llc_array_conversion.py.  Maybe you can make the fix and pull request.   

Presumably the test code can look in both directories:
./ecco-v4-py/binary_data/ 
/home/username/binary_data 

For the required files.

**Comment by timothyas** [comment]
I wrote the test to look for necessary files, otherwise skip the test. See discussion at #74. I'll close this issue once we handle #74 since I think this will taken care of.

**Comment by ifenty** [comment]
@timothyas 

I see that when installing the package from pip the /binary_data directory goes at the head of the environment.  So, when I use an environment called 'ecco' in anaconda, it goes to

`` /home/ifenty/anaconda3/envs/ecco/binary_data``

and when I don't use anaconda it goes to 

`` /home/ifenty/binary_data``

In anaconda the environment variable is  ``CONDA_PREFIX``:
``declare -x CONDA_PREFIX="/home/ifenty/anaconda3/envs/ecco"``

so, I guess what I'm saying is that finding the binary_data directory may not be easy.  

I'd say first look in the local ecco-v4-py directory (hoping that the package was installed as  github clone)  then if it's not there, maybe prompt the user for it.

**Comment by timothyas** [comment]
Thanks @ifenty, see my updated comment in #74

**Comment by timothyas** [comment]
Hey all, I think this was addressed so I'll go ahead and close it.
